### DISCUSSION TITLE
Cannot create ApplicationGateway

### DISCUSSION THREAD

**Discussion by FairW** [Azure/ResourceModules#4111] [Q&A]
I'm struggling to setup a valid bicep for ApplicationGateway using  bicep/modules/microsoft.network.applicationgateways . 

I get this error in my devops pipeline:
```
"code": "InvalidRequestFormat",
"message": "System.ArgumentNullException: Value cannot be null.\r\nParameter name: groupName"
```

...but nothing tells what params I send to the appgw module generates the problem... I don't see anything called `groupName `in the template.

Also; there are no more error details in the portal - deploy target Resource Group - deployment blade:

```
"statusCode": "BadRequest",
        "statusMessage": "{\"error\":{\"code\":\"InvalidRequestFormat\",\"message\":\"Cannot parse the request.\",\"details\":[{\"code\":\"InvalidRequestFormat\",\"message\":\"System.ArgumentNullException: Value cannot be null.\\r\\nParameter name: groupName\"},{\"code\":\"InvalidRequestFormat\",\"message\":\"System.ArgumentNullException: Value cannot be null.\\r\\nParameter name: groupName\"}]}}",
        "eventCategory": "Administrative",
```

-Any tip?

**Comment by marcvoor** [comment]
Ever found out what the problem was, because I am bumping into the same problem here.

Thanks.

**Reply by marcvoor** [reply]
OK, for me the issue was that I did not use the subscription id and resource group name when using the resourceid() method in bicep. After adding these, the problem was gone.

**Reply by Henry-SoHandsome** [reply]
thanks, dude~! you help a lot

**Comment by aavdberg** [comment]
Can confirm that this for me also worked, hat this problem with the AVM module for Application Gateway.

https://github.com/Azure/bicep-registry-modules/issues/4272
